src/condor_schedd.V6/schedd.cpp

```diff
@@ -10724,14 +10862,29 @@ Scheduler::start_sched_universe_job(PROC_ID* job_id)
 		}
 	}
 
-		// Scheduler universe jobs should not be told about the shadow
-		// command socket in the inherit buffer.
-	daemonCore->SetInheritParentSinful( NULL );
+	GetAttributeBool(job_id->cluster, job_id->proc, ATTR_IS_DAEMON_CORE, &is_daemon_core);
+	if (is_daemon_core) {
+		auto public_part = std::unique_ptr<char, decltype(free)*>{Condor_Crypt_Base::randomHexKey(), free};
+		auto private_part = std::unique_ptr<char, decltype(free)*>{Condor_Crypt_Base::randomHexKey(), free};
+		if ( ! public_part.get() || ! private_part.get()) {
+			dprintf(D_ERROR, "Failed to create secret for scheduler universe Daemon.\n");
+			goto wrapup;
+		} else {
+			ClaimIdParser cidp(public_part.get(), nullptr, private_part.get());
+			cmd_secret = cidp.claimId();
+			envobject.SetEnv(ENV_CONDOR_SECRET, cmd_secret.c_str());
+		}
```

Comment:
DAGMan and its clients are using this as an opaque secret string. That means it can be just a random sequence of characters. You don't need to construct a ClasIdParser.